Adopt mCoding Project Structure

beauremus opened the pull request:
I came across this video that describes a best practice structure and workflow for a Python package.
https://youtu.be/DhUpxWjOhME
I used the example repo to help get most of the way there.
https://github.com/mCodingLLC/SlapThatLikeButton-TestingStarterProject

Pytest is compatible with unittest but I want to go back and use new features.
Also, I want to use mypy.

beauremus committed:
refactor: Use standard tests naming convention.

Also, refer to files using pathlib.

beauremus committed:
style: Enable .vscode linting.

beauremus committed:
refactor: Remove unneeded copy paste.

beauremus committed:
refactor: Use project structure from mCoding.

Using the Python project structure recommended by mCoding.
mCodingLLC/SlapThatLikeButton-TestingStarterProject on GitHub.

beauremus committed:
fix: Remove unneeded and broken coverage report.

beauremus committed:
fix: Refer to correct badge.

I copied the badge from mCoding. It's now referring to our repo.

julianbadillo commented on .vscode/settings.json:
```diff
@@ -0,0 +1,4 @@
+{
```
You added this so linting is enforced?
It kind of broke my unit testing through the VSCode plugin, maybe we could include those settings here also?
```json
"python.testing.unittestEnabled": true
```

beauremus replied:
Yes. Merge that in as well.

beauremus replied:
I will add it.

beauremus replied:
I enabled `"python.testing.pytestEnabled": true` since we're using pytest now.

julianbadillo commented on README.md:
````diff
@@ -124,11 +126,21 @@ To test local modifications, use pip's editable mode.
 
 ```bash
-coverage run -m unittest discover
+pytest
````
When I type `pytest` I get this error:
```bash
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=drf_generator
  inifile: C:\Users\jbadillo\Documents\workspace\drf-generator\pyproject.toml
  rootdir: C:\Users\jbadillo\Documents\workspace\drf-generator
```

julianbadillo replied:
Also, to run unit tests I need to install it as a package (with the `pip install -e .` command), otherwise it would complain:
```bash
Traceback (most recent call last):
  File ".\tests\drf_generator_test.py", line 7, in <module>
    import drf_generator.generator as drf_generator
ModuleNotFoundError: No module named 'drf_generator'
```
Both on the terminal and VS Code unit testing plugin.

julianbadillo replied:
Nevermind, I was missing `pytest-cov` in my environment. This works well on terminal.

julianbadillo commented on README.md:
````diff
@@ -124,11 +126,21 @@ To test local modifications, use pip's editable mode.
 ### Generate coverage report
 
 ```bash
 coverage report -m
````
This won't show anything unless `coverage` is run beforehand.

julianbadillo replied:
Nevermind, after running `pytest`, the coverage report is generated.

beauremus committed:
test: Add VSCode pytestEnabled flag.

julianbadillo reviewed (approved):
Unit tests work in the terminal and VSCode plugin fine.
The package needs to be installed in pip to run them, though.

beauremus committed:
build: Clarify testing process with Make.

Add docs to the README.md file to describe testings process and use
Make to simplify the process.

julianbadillo commented:
@beauremus I'm merging this.
